### PR TITLE
Upgrade react-scripts to 4.0.2

### PR DESCRIPTION
**Pull request by axelboc**
https://github.com/facebook/create-react-app/releases/tag/v4.0.2

### CODE REVIEW

File: .env [hunk 1]
```diff
@@ -21,6 +21,10 @@ REACT_APP_JLAB_FALLBACK_DOMAIN="water_224.h5"
 # Rendering performance profiling (local development only)
 REACT_APP_PROFILING_ENABLED=false
 
+# Disable ESLint error modal in development (log warnings instead)
+ESLINT_NO_DEV_ERRORS=true
```
Comment (axelboc) [inline .env:25]: Yay! No more bloody error modal for an unused var!! Note that TS errors will still bring up a modal.

File: package.json [hunk 2]
```diff
@@ -101,11 +101,12 @@
     "prettier": "^2.2.1",
     "react-app-polyfill": "^2.0.0",
     "react-is": "^16.13.1",
-    "react-scripts": "4.0.1",
+    "react-scripts": "4.0.2",
     "typescript": "^4.1.3",
     "worker-loader": "^3.0.7"
   },
   "jest": {
+    "restoreMocks": true,
```
Comment (axelboc) [inline package.json:109]: Avoids mocks leaking into other tests.

File: .env (hunk 1 above)
Comment (axelboc) [reply .env:25]: Well, bad example, since we turn unused vars errors into warnings in our ESLint config, but you know what I mean.

File: .env [hunk 3]
```diff
@@ -21,6 +21,10 @@ REACT_APP_JLAB_FALLBACK_DOMAIN="water_224.h5"
...
+
 # Logging limit of React Testing Library
-# https://testing-library.com/docs/dom-testing-library/api-helpers/#debugging
+# Don't override this variable in `.env.local`, as this file is not used during tests
```
Comment (loichuder) [inline .env:28]: This means it should be set in `.env` to have an effect ?

**Review by loichuder — APPROVED** [review]

File: .env (hunk 3 above)
Comment (axelboc) [reply .env:28]: You can set it temporarily in `.env` and not commit it, on the terminal directly (before `npm test`), or in `.env.test.local` if you want to keep it around.

**Pull request merged**